### PR TITLE
hwmv2 boards: arm: Add support for WeAct Studio MiniSTM32H743 Core board

### PR DESCRIPTION
**Pull request by CharlesDias**
Adds the device trees, Kconfig, and documentation files.

The following features have been confirmed working on hardware:
* LED
* Button
* USB CDC ACM
* External QuadSPI NOR Flash memory
* External SPI NOR Flash memory
* SDMMC
* TFT-LCD 160x80 pixel

The following sample codes were used to validate:
* samples/basic/blinky
* samples/basic/button/
* samples/hello_world
* samples/subsys/shell/shell_module/
* samples/drivers/spi_flash
* samples/drivers/flash_shell/
* samples/subsys/display/lvgl

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]
Zephyr is migrating to hwmv2 and will not be accepting hwmv1 boards or socs after this change, this submission should be converted to hwmv2

**Comment by nordicjm** [comment]
> > Zephyr is migrating to hwmv2 and will not be accepting hwmv1 boards or socs after this change, this submission should be converted to hwmv2
> 
> Hi, @nordicjm. Do I have to migrate and submit the PR to [collab-hwm](https://github.com/zephyrproject-rtos/zephyr/tree/collab-hwm) branch or waiting until [collab-hwm](https://github.com/zephyrproject-rtos/zephyr/tree/collab-hwm) goes to upstream?
> 
> Thanks.

it would need to be submitted after the completion of the hwmv2 migration, at which point it would be submitted against `main`

**Comment by CharlesDias** [comment]
> Zephyr is migrating to hwmv2 and will not be accepting hwmv1 boards or socs after this change, this submission should be converted to hwmv2

Hi @nordicjm. The PR was updated to hwmv2. Let me know if any other modifications are needed. Thanks.

File: boards/weact/mini_stm32h743/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# WeAct Studio MiniSTM32H7 board configuration
+
+# Copyright (c) Charles Dias <charlesdias.cd@outlook.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_MINI_STM32H743
+
+config BOARD
+	default "mini_stm32h743"
+
```
Comment (nordicjm) [inline boards/weact/mini_stm32h743/Kconfig.defconfig:10]:
```suggestion
```

**Comment by nordicjm** [comment]
This will need to be re-targeted to `main` once hwmv2 is merged, it can then be merged after that point

File: boards/weact/mini_stm32h743/Kconfig.defconfig (hunk 1 above)
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/Kconfig.defconfig]: Done.

**Comment by CharlesDias** [comment]
Fix color setting.

rect15774

**Comment by nordicjm** [comment]
Needs rebase

**Comment by nordicjm** [comment]
And changing to `main` branch

**Comment by CharlesDias** [comment]
> And changing to `main` branch

Dear, @nordicjm. Requested changes have been made.

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/weact/mini_stm32h743/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# WeAct Studio MiniSTM32H7 board configuration
+
+# Copyright (c) Charles Dias <charlesdias.cd@outlook.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_MINI_STM32H743
+
+config BOARD
+	default "mini_stm32h743"
```
[the remark this answers is not in the document]
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/Kconfig.defconfig]: Done. Thanks!

File: boards/weact/mini_stm32h743/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+if BOARD_MINI_STM32H743
+
+config INPUT
+	default y if LVGL
+
+config LV_COLOR_16_SWAP
+	default y if LVGL
+
+config LV_Z_BITS_PER_PIXEL
+	default 16 if LVGL
```
Comment (erwango) [inline boards/weact/mini_stm32h743/Kconfig.defconfig:15]: Put those under `if DISPLAY`

File: boards/weact/mini_stm32h743/mini_stm32h743.dts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+	model = "WeAct Studio MiniSTM32H743 Core Board";
+	compatible = "weact,mini-stm32h743";
+
+	chosen {
+		zephyr,console = &usb_cdc_acm_uart;
+		zephyr,shell-uart = &usb_cdc_acm_uart;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,display = &st7735r_tft;
+		sdhc0 = &sdmmc1;
```
Comment (erwango) [inline boards/weact/mini_stm32h743/mini_stm32h743.dts:22]: sdhc0 is an alias, not a chosen

File: boards/weact/mini_stm32h743/mini_stm32h743_defconfig [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+CONFIG_CLOCK_CONTROL=y
+
+# Enable pin controller
+CONFIG_PINCTRL=y
+
+# Enable SPI
+CONFIG_SPI=y
+
+# Enable SPI driver for ST7735R
+CONFIG_ST7735R=y
```
Comment (erwango) [inline boards/weact/mini_stm32h743/mini_stm32h743_defconfig:29]: This can't be part of default board configuration

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/weact/mini_stm32h743/mini_stm32h743.dts (hunk 4 above)
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/mini_stm32h743.dts]: Done

File: boards/weact/mini_stm32h743/Kconfig.defconfig (hunk 3 above)
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/Kconfig.defconfig]: Done. Also, I added the `LV_COLOR_16_SWAP` and `LV_Z_BITS_PER_PIXEL` under `if LVGL`

File: boards/weact/mini_stm32h743/mini_stm32h743_defconfig (hunk 5 above)
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/mini_stm32h743_defconfig]: Fixed. I removed and create the `samples/subsys/display/lvgl/boards/mini_stm32h743.conf` file to work with the LVGL sample.

Thanks for your review!

File: samples/subsys/display/lvgl/boards/mini_stm32h743.conf [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) Charles Dias <charlesdias.cd@outlook.com>
+# SPDX-License-Identifier: Apache-2.0
+
+# Enable SPI
+CONFIG_SPI=y
```
Comment (erwango) [inline samples/subsys/display/lvgl/boards/mini_stm32h743.conf:5]: According to https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/display/Kconfig.st7735r#L10, this line is not required.

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: samples/subsys/display/lvgl/boards/mini_stm32h743.conf (hunk 6 above)
Comment (CharlesDias) [reply samples/subsys/display/lvgl/boards/mini_stm32h743.conf]: Removed.

**Review by erwango — DISMISSED** [review]

File: boards/weact/mini_stm32h743/doc/img/stm32h7xx.webp
Comment (nordicjm) [inline boards/weact/mini_stm32h743/doc/img/stm32h7xx.webp:1]: Can you reduce the size of this image? We don't need a 1200x1200 image, 384x384 would be more than enough

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
Actually it doesn't really seem any different, 83KiB is **huge**, can you run it through something like https://tinypng.com/ ?

**Comment by CharlesDias** [comment]
@nordicjm, the final size now is 26.3 kB. Thanks for your tip.

**Comment by kartben** [comment]
> @nordicjm, the final size now is 26.3 kB. Thanks for your tip.

ah! there you go :)

**Review by nordicjm — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Comment by CharlesDias** [comment]
Dear @nordicjm, @erwango, and @kartben,

Do I need to do anything else in this PR?
Thanks.

**Comment by erwango** [comment]
@kartben PTAL

File: boards/weact/mini_stm32h743/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+
+The MiniSTM32H743 Core board is configured as follows
+
+- USER_LED : PE3
+- USER_PB : PC13
+- SPI1 SCK/MISO/MOSI/NSS : PB3/PB4/PD7/PD6 (NOR Flash memory)
+- SPI4 SCK/MOSI/NSS : PE12/PE14/PE11 (LCD)
+- QuadSPI CLK/NCS/IO0/IO1/IO2/IO3 : PB2/PB6/PD11/PD12/PE2/PD13 (NOR Flash memory)
+- SDMMC1 CLK/DCMD/CD/D0/D1/D2/D3 : PC12/PD2/PD4/PC8/PC9/PC10/PC11 (microSD card)
+- USB DM/DP : PA11/PA12 (USC CDC ACM)
```
Comment (kartben) [inline boards/weact/mini_stm32h743/doc/index.rst:131]:
```suggestion
- USB DM/DP : PA11/PA12 (USB CDC ACM)
```

File: boards/weact/mini_stm32h743/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: mini_stm32h743
+   :goals: build flash
+   :gen-args: -DCONFIG_BOOT_DELAY=5000
+
+You can build and flash the examples to make sure Zephyr is running correctly on
+your board. The LED definitions can be found in
+:zephyr_file:`boards/arm/mini_stm32h743/mini_stm32h743.dts`.
```
Comment (kartben) [inline boards/weact/mini_stm32h743/doc/index.rst:218]:
```suggestion
:zephyr_file:`boards/weact/mini_stm32h743/mini_stm32h743.dts`.
```

**Review by kartben — CHANGES_REQUESTED** [review]
a couple minor-ish typos - thanks!

File: boards/weact/mini_stm32h743/doc/index.rst (hunk 7 above)
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/doc/index.rst]: Fixed.

File: boards/weact/mini_stm32h743/doc/index.rst (hunk 8 above)
Comment (CharlesDias) [reply boards/weact/mini_stm32h743/doc/index.rst]: Fixed. Thanks.

**Review by kartben — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**